qa/vagrant/build.gradle

```diff
@@ -69,6 +71,24 @@ repositories {
 }
 
 dependencies {
+  String maybeTestsSeed = System.getProperty("tests.seed", null);
+  if (maybeTestsSeed != null) {
+    seed = new BigInteger(maybeTestsSeed, 16).longValue()
+    formattedSeed = maybeTestsSeed
+  } else {
+    seed = new Random().nextLong()
+    formattedSeed = String.format("%016X", seed)
+  }
+
+  String maybeUpdradeFromVersions = System.getProperty("tests.packaging.upgrade.from.versions", null)
+  if (maybeUpdradeFromVersions != null) {
+    upgradeFromVersions = maybeUpdradeFromVersions.split(",")
+  } else {
+    upgradeFromVersions = new File(project.projectDir, 'versions')
+  }
+
+  upgradeFromVersion = upgradeFromVersions[new Random(seed).nextInt(upgradeFromVersions.length)]
```

Comment:
Putting these in the dependencies closure doesn't really do anything. I'd probably just move them back up to the top.

I don't think we have any chance of avoiding the IO on startup given how gradle handles depdencies. I think it is worth it so we can have this randomization though.